cmd/derper/derper.go

```diff
@@ -44,7 +44,8 @@ var (
 	versionFlag = flag.Bool("version", false, "print version and exit")
 	addr        = flag.String("a", ":443", "server HTTP/HTTPS listen address, in form \":port\", \"ip:port\", or for IPv6 \"[ip]:port\". If the IP is omitted, it defaults to all interfaces. Serves HTTPS if the port is 443 and/or -certmode is manual, otherwise HTTP.")
 	httpPort    = flag.Int("http-port", 80, "The port on which to serve HTTP. Set to -1 to disable. The listener is bound to the same IP (if any) as specified in the -a flag.")
-	stunPort    = flag.Int("stun-port", 3478, "The UDP port on which to serve STUN. The listener is bound to the same IP (if any) as specified in the -a flag.")
+	stunAddr    = flag.String("stun-addr", "", "server UDP listen address, in form \"ip\", or for IPv6 \"[ip]\". If left empty, defaults to the address specified in the -a flag. The port *must* be provided separately as -stun-port due to backwards compatibility.")
+	stunPort    = flag.Int("stun-port", 3478, "The UDP port on which to serve STUN. The listener is bound to the IP specified in the -stun-addr flag or the -a flag.")
```

Comment:
what if we instead changed --stun-port to be a String type, and let it be purely numeric to mean a port number only (today's behavior) or if it successfully passes, net.SplitHostPort, then do the new behavior? Then we don't have a proliferation of flags and associated documentation burden.

The downside, of course, is the flag is named a bit oddly, but there's already some inconsistencies in these flag names so it's not the end of the world.

Reply:
OK, sure! Done.

```
❯ doas ./derper
2024/07/02 20:34:22 STUN server listening on [::]:3478
2024/07/02 20:34:22 derper: serving on :443 with TLS
❯ doas ./derper -stun-port 6969
2024/07/02 20:34:26 STUN server listening on [::]:6969
2024/07/02 20:34:26 derper: serving on :443 with TLS
❯ doas ./derper -stun-port :6969
2024/07/02 20:34:29 STUN server listening on [::]:6969
2024/07/02 20:34:29 derper: serving on :443 with TLS
❯ doas ./derper -stun-port localhost:6969
2024/07/02 20:34:32 STUN server listening on 127.0.0.1:6969
2024/07/02 20:34:32 derper: serving on :443 with TLS
```